solutions/devsprint-pedro-menezes-1/DeliveryAppChallenge/Modules/Product/Scenes/GabrielaSillis/GSLoginViewController.swift

```diff
@@ -223,3 +176,84 @@ extension GSLoginViewController {
         loginButton.isEnabled = true
     }
 }
+
+private extension GSLoginViewController {
+    func configureTextFieldWithDefaultValue(
+        emailTextField: String = "clean.code@devpass.com",
+        passwordTextField: String = "111111") {
+        self.emailTextField.text = emailTextField
+        self.passwordTextField.text = passwordTextField
+    }
+    
+    func verifyIfUserIsAlreadyLoggedIn() {
+        if let _ = UserDefaultsManager.UserInfos.shared.readSesion() {
+            coordinator.startNavigatingFlow()
+        }
+    }
+    
+    func checkDeviceConnectivityAnduRequestAuthentication() {
+        let deviceDoNotHaveConnectivity = !ConnectivityManager.shared.isConnected
+        if deviceDoNotHaveConnectivity {
+            Globals.showNoInternetCOnnection(
+                controller: self)
+            return
+        }
+        viewModel.makeLoginAuthenticationRequest()
+    }
+    
+    func sendTextFieldDataToViewModel() {
+        let email = emailTextField.text ?? ""
+        let password = passwordTextField.text ?? ""
+        viewModel.getUserEmailAndPasswordTextField(email: email, password: password)
+    }
+    
+    func showErrorMessage() {
+        self.setErrorLogin("E-mail ou senha incorretos")
+        coordinator.displayErrorAlertView()
+    }
+    
+    func shouldShowOrHiddenPasswordInput() {
+        if showPassword {
+            isPasswordTextFieldSecureTextEntry(false)
+            configureImageButton(imageName: "eye.slash")
+        } else {
+            isPasswordTextFieldSecureTextEntry(true)
+            configureImageButton(imageName: "eye")
+        }
+    }
+    
+    func isPasswordTextFieldSecureTextEntry(_ isSecureTextEntry: Bool) {
+        passwordTextField.isSecureTextEntry = isSecureTextEntry
+    }
+    
+    func configureImageButton(imageName: String) {
+        showPasswordButton.setImage(UIImage.init(systemName: imageName)?.withRenderingMode(.alwaysTemplate), for: .normal)
+    }
+}
+
+extension GSLoginViewController: GSLoginViewModelDelegate {
+    func successLoginAuthenticationRequest(session: Session) {
```

Comment:
funçao sempre começa com verbo
tente trabalhar com duas palavras no parâmetro (lembre do `with`, `for`, `of` ?)
para não repetir a palavra session na hora que for usar a funçao

Reply:
boat